src/test/scala/chiselTests/PrintableSpec.scala

```diff
@@ -47,87 +49,86 @@ class PrintableSpec extends AnyFlatSpec with Matchers {
         case _ => fail(s"Regex to process Printf should work on $str!")
       }
     }
-
     firrtl.split("\n").collect {
       case PrintfRegex(matched) =>
         val (str, args) = processBody(matched)
         Printf(str, args)
     }
   }
 
+  // Generates firrtl, gets Printfs
+  // Returns None if failed match; else calls the partial function which could have its own check
+  // Returns Some(true) to caller
+  // Not calling fail() here - and letting caller do so - helps in localizing errors correctly.
+  private def generateAndCheck(gen: => RawModule, pos: Position)(check: PartialFunction[Seq[Printf], Unit]) = {
+    val firrtl = ChiselStage.emitChirrtl(gen)
+    val printfs = getPrintfs(firrtl)
+    if (!check.isDefinedAt(printfs)) {
+      fail()(pos)
```

Comment:
```suggestion
  private def generateAndCheck(gen: => RawModule)(check: PartialFunction[Seq[Printf], Unit])(implicit pos: Position) = {
    val firrtl = ChiselStage.emitChirrtl(gen)
    val printfs = getPrintfs(firrtl)
    if (!check.isDefinedAt(printfs)) {
      fail()
```
You should not need to pass `pos` manually, if you use an implicit here, you can get rid of `Position.here` at every call site.

Reply:
@jackkoenig done!